docs/admin/k8s-deployment.md

```diff
@@ -1,172 +1,404 @@
+installation](#verify-the-installation) below to make sure it worked.
+
+### Quick Start with kind and the bpfd-operator
+
+Using kind with the bpfd-operator is an easy way to spin up a local kind cluster
+and deploy bpfd.
+
+#### Pre-requsites
+
+- Install `git`, `make`, and `go`
+- Install [Docker](https://docs.docker.com/engine/install/) for your platform.
+  - We also recommend that you follow the [Post-installation steps for
+    Linux](https://docs.docker.com/engine/install/linux-postinstall/) to allow
+    your user to run Docker commands without root priviledges.
+- Install [kubctl](https://kubernetes.io/docs/reference/kubectl/kubectl/) and
+  [kind](https://kind.sigs.k8s.io/)
+  - [Installation Instructions](https://kubernetes.io/docs/tasks/tools/)
+- clone the bpfd repository
 
 ```bash
-cargo run --bin crdgen | kubectl apply -f -
+git clone https://github.com/redhat-et/bpfd.git
+cd bpfd/bpfd-operator
 ```
 
-2. Install CA and Necessary certs.
+The bpfd-operator was built utilizing some great tooling provided by the
+[operator-sdk library](https://sdk.operatorframework.io/). A great first step in
+understanding some of the functionality can be to cd into the
+`bpfd/bpfd-operator` directory and run `make help`.
+
+#### Deploy locally via KIND
+
+After reviewing the possible make targets it's quick and easy to get bpfd
+deployed locally on your system via a [KIND cluster](https://kind.sigs.k8s.io/).
+with:
 
 ```bash
-kubectl apply -f ./packaging/kubernetes-deployment/bpfd-core/bpfd-cert-issuer.yaml
-kubectl apply -f ./packaging/kubernetes-deployment/bpfd-core/bpfd-certs.yaml
+make run-on-kind
 ```
 
-**Note:** There may be a delay in cert-manager being completely up, even if pod state is running.
-If applying `bpfd-cert-issuer.yaml` fails with something like `x509: certificate signed by unknown
-authority`, check the `cert-manager` logs to verify that is it completely ready.
+> **_NOTE:_**  You will need a spare 20G in your root partition to run `make
+> run-on-kind`, and it will take a while the first time you run it.
 
-3. Install `bpfd` configmap.
-   This configmap contains the `bpfd.toml` file content (see [configuration.md](configuration.md)).
-   When the bpfd daemonset is created in step 5, this configmap will be propagated to each node via a volume mount in the bpfd container.
+By default local image builds will be used for the kind deployment, however, the
+container images used for `bpfd`,`bpfd-agent`, and `bpfd-operator` can also be
+configured as follows.
 
 ```bash
-kubectl apply -f ./packaging/kubernetes-deployment/bpfd-core/bpfd-config.yaml
+BPFD_IMG=<your/image/url> BPFD_AGENT_IMG=<your/image/url> BPFD_OPERATOR_IMG=<your/image/url> make run-on-kind
 ```
 
-4. Install bpfd `ServiceAccount`, `ClusterRole`, and `ClusterRoleBinding`.
+To push and test out any local changes simply run:
 
 ```bash
-kubectl apply -f ./packaging/kubernetes-deployment/bpfd-core/bpfd-agent-rbac.yaml
+make kind-reload-images
 ```
 
```

Comment:
Add something like:

To tear down the kind cluster, run:

```bash
kind delete cluster -n bpfd-deployment
```

The default name of the created cluster is `bpfd-deployment`.
To retrieve all the created kind clusters, use `kind get clusters`.